docs/ota-client-guide/modules/ROOT/pages/garage-sign.adoc

```diff
@@ -0,0 +1,256 @@
++
+[source,bash]
+----
+cd garage-sign/
+bin/garage-sign --help
+----
+
+For more information, please refer to the link:https://github.com/advancedtelematic/ota-tuf/tree/master/cli[`garage-sign` README file].
+
+
+== Use of `garage-sign` with OTA Connect
+
+To use the `garage-sign` tool with OTA Connect, you need to perform the following steps:
+
+. Initialize a new image repository locally on your computer that pulls your `credentials.zip` from OTA Connect.
++
+[source, bash]
+----
+garage-sign init \
+  --repo <reponame> \ <1>
+  --credentials </path/to/credentials.zip> <2>
+----
++
+<1> The name of your image repository which is to be initialized by the `init` command is provided here.
+<2> Here you provide the path of your `credentials.zip` file downloaded from your account on the https://connect.ota.here.com[OTA Connect Portal].
+
+. Ensure you have the latest `targets.json` file by pulling it from OTA Connect.
++
+[source, bash]
+----
+garage-sign targets pull --repo <reponame>
+----
+
+The above commands create a local repository in a directory called `/tuf/<reponame>`. This directory will contain your `credentials.zip` file and has a tree structure as shown below:
+
+[source, bash]
+----
+tuf/<reponame>
+├── config.json
+├── credentials.zip
+├── keys
+│   ├── targets.pub <1>
+│   └── targets.sec <2>
+└── roles
+    ├── root.json   <3>
+    ├── targets.json.checksum
+    └── unsigned
+        └── targets.json <4>
+----
+
+<1> The public part of your `targets` key
+<2> The private part of your `targets` key
+<3> The `root.json` metadata file
+<4> The `targets.json` metadata file
+
+
+== Use cases of `garage-sign` with OTA Connect
+
+In most use cases, more than one `garage-sign` command is used to perform a specific task. A few use cases will be discussed in this section, however the use of the `garage-sign` tool is not limited to these.
+
+=== Generate new keys
+
+You might generate new keys if you:
+
+* xref:rotating-signing-keys.adoc#_rotate_the_online_keys_with_your_new_offline_keys[want to rotate your online keys with your new offline keys].
+* believe your keys have been compromised.
+* would like to revoke the current keys for signing metadata.
+
+==== Workflow to generate new keys
+
+When generating new keys for metadata, you perform the following steps:
+
+. First, generate the new `root` and `targets` keys by running the command `garage-sign key generate` twice.
++
+[source,bash]
+----
+garage-sign key generate \
+  --repo <reponame> \
+  --name myroot \ <1>
+  --type <ed25519|rsa> <2>
+garage-sign key generate \
+  --repo <reponame> \
+  --name mytargets \
+  --type <ed25519|rsa>
+----
++
+<1> You can give any name to the keys you generate, which helps to easily identify between the new `root` and `targets` keys. Here, the keys generated are given the names `myroot` and `mytargets`.
+<2> The key type must be specified as either Ed25519 or RSA. In OTA Connect, an RSA key is used for signing uptane metadata.

```

Comment:
Why can I specify ed25519 here, if I can only use RSA?

Reply:
@tkfu good point. The help specified both that's why I wrote both within the syntax, but since only RSA is used for OTA I'll remove the option for ed25519.

Reply:
AFAIK ed25119 is supposed to work. However, I think @kbushgit said it did not the last time he tried. Can someone confirm if it does work? @simao perhaps?